Incorporate the place for new_module_config.xml missing in regression-report.yml

## Restore New Module Config Functionality in Regression Report Workflow

### Background
During the migration from `diff-report.yml` to `regression-report.yml` and the transition from Groovy to Java for the diff tool, we inadvertently omitted the functionality to generate reports for new modules which was a part of `diff-report.yml`.

### Current Problem
The `regression-report.yml` workflow lacks the ability to properly handle and execute the new module config report generation, despite having the conditional logic in place:

```yaml
if [ -f new_module_config.xml ]; then
  echo "Generating report for new module"
  java -jar difftool.jar -r "$REPO" -p "$PR_BRANCH" -c new_module_config.xml -m single \
    -l project.properties -xm "-Dcheckstyle.failsOnError=false" \
    --allowExcludes
```

### Required Changes
1. Update the `regression-report.yml` workflow to properly download, cache, and utilize the `new_module_config.xml` file when present.
2. Ensure that the Java-based diff tool supports the same functionality as the previous Groovy implementation for new module configs.
4. Add logging for this specific case.

### Expected Outcome
After implementation, the workflow should be able to:
1. Detect the presence of a new module config in the PR description.
2. Download and use the `new_module_config.xml` file correctly.
3. Generate a report specific to the new module using the provided configuration.

This will restore parity with the previous workflow's capabilities and ensure that new module additions can be properly tested and reported on.

### Present scenario and Detection
A sample PR description would look like below:
```
New module config: https://gist.githubusercontent.com/kalpadiptyaroy/187788693b6445aed065ce39a171c502/raw/cdb3feb926b07fd8c07df72997df9de4d38957ee/checks.xml

Diff Regression projects: https://gist.githubusercontent.com/Zopsss/9960953f88e763d31eea3a14447b861f/raw/edfd96ea7a5757cea2e08856be98776cbb21a171/guava-project.properties
```

At present, if i use the existing workflow and above PR description, it fails with below error message:
```
Run cd .ci-temp/test-configs/checkstyle-tester
Generating standard diff report
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Error: Process completed with exit code 1.
```

Detected at PR: https://github.com/checkstyle-gsoc/checkstyle/pull/29
Detected at Runner:  https://github.com/checkstyle-gsoc/checkstyle/actions/runs/10931106285/job/30345435288

--- a/.github/workflows/regression-report.yml
+++ b/.github/workflows/regression-report.yml
@@ -318,6 +318,7 @@ jobs:
       projects_link: ${{ steps.parse_description.outputs.projects_link }}
       report_label: ${{ steps.parse_description.outputs.report_label }}
       patch_config_link: ${{ steps.parse_description.outputs.patch_config_link }}
+      new_module_config_link: ${{ steps.parse_description.outputs.new_module_config_link }}
     steps:
       - name: Restore cache
         uses: actions/cache/restore@v4
@@ -401,6 +402,10 @@ jobs:
             }}
           PATCH_CONFIG_LINK: ${{
             needs.handle_configs_in_pr_description.outputs.patch_config_link
+            || ''
+            }}
+          NEW_MODULE_CONFIG_LINK: ${{
+            needs.handle_configs_in_pr_description.outputs.new_module_config_link
             || ''
             }}
           DIFF_PROJECTS_LINK: ${{
@@ -523,6 +528,17 @@ jobs:
             echo "patch_config.xml not found"
           fi
           echo "-----------------------------"
+
+      - name: Print new_module_config.xml
+        run: |
+          echo "Contents of new_module_config.xml:"
+          echo "-----------------------------------"
+          if [ -f .ci-temp/new_module_config.xml ]; then
+            cat .ci-temp/new_module_config.xml
+          else
+            echo "new_module_config.xml not found"
+          fi
+          echo "-----------------------------------"
 
       - name: Generate report
         run: |

